File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;    

        for (int k = 0; k < nums.size(); k++)
        {
            map[nums[k]] = k;
        }

        for (int i = 0; i < nums.size(); i++)
        {
            if (map.count(target - nums[i]) && map[target - nums[i]] != i)
            {
                return {i, map[target - nums[i]]};
            }
        }
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int left = 0;
        int right = nums.Length - 1;
        
        while (left <= right)
        {
            int m = (left + right) / 2;
            if (nums[m] == target)
                return m;
            else if (nums[m] < target)
            {
                left = m + 1;
            }
            else {
                right = m - 1;
            }
        }

        return -1;

    }
}
